server/datastore/mysql/query_results.go

```diff
@@ -0,0 +1,65 @@
+package mysql
+
+import (
+	"context"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	// QueryResultRowLimit is the maximum number of rows that can be stored per query
+	QueryResultRowLimit = 1000
+)
+
+func (ds *Datastore) SaveQueryResultRow(ctx context.Context, row *fleet.ScheduledQueryResultRow) (*fleet.ScheduledQueryResultRow, error) {
+	insertStmt := `
+		INSERT INTO query_results (query_id, host_id, last_fetched, data)
+			VALUES (?, ?, ?, ?)
+		`
+	_, err := ds.writer(ctx).ExecContext(ctx, insertStmt, row.QueryID, row.HostID, row.LastFetched, row.Data)
+	if err != nil && isDuplicate(err) {
```

Comment:
Under what scenarios can this `INSERT` can fail with a duplicate? (I don't see any unique keys but maybe am missing something)